### PR TITLE
feat: implement s3-base protocol download provider

### PR DESCRIPTION
**Pull request by frrist**
- supported via bacalhau get assuming credentals are present in env vars or aws config file
- closes https://github.com/bacalhau-project/expanso-planning/issues/304

### CODE REVIEW

**Review by rossjones — DISMISSED** [review]
LGTM.
Couple of comments that you might want to address, and I don't think this handles the case where presigned urls are disabled, but the user chose compression in the publisher.  Should we do anything special in that case (such as unzipping the download)?

File: pkg/downloader/s3/download.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	"github.com/bacalhau-project/bacalhau/pkg/downloader"
+	s32 "github.com/bacalhau-project/bacalhau/pkg/s3"
+)
+
+type Downloader struct {
+	provider *s32.ClientProvider
+}
+
+func (d *Downloader) IsInstalled(ctx context.Context) (bool, error) {
```
[the remark this answers is not in the document]
Comment (frrist) [reply pkg/downloader/s3/download.go:20]: You're right, I need to check if the embedded provider is install, good catch

File: pkg/s3/utils.go [hunk 2]
```diff
@@ -19,8 +19,7 @@ func DefaultAWSConfig() (aws.Config, error) {
 			return aws.Config{}, err
 		}
 	}
-	var optFns []func(*config.LoadOptions) error
-	return config.LoadDefaultConfig(context.Background(), optFns...)
+	return config.LoadDefaultConfig(context.Background())
```
[the remark this answers is not in the document]
Comment (frrist) [reply pkg/s3/utils.go:22]: No, it has some defaults: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html
> Environment Variables
> By default, the SDK detects AWS credentials set in your environment and uses them to sign requests to AWS. That way you don’t need to manage credentials in your applications.
The SDK looks for credentials in the following environment variables:
    AWS_ACCESS_KEY_ID
    AWS_SECRET_ACCESS_KEY
    AWS_SESSION_TOKEN (optional)

Which is great if you are running on aws, and a PITA if you are running in another cloud..

File: pkg/downloader/s3/download.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	// Write the contents of S3 Object to the file
+	if _, err := dwnld.Download(ctx, f, &s3.GetObjectInput{
+		Bucket: &bucket,
+		Key:    &key,
+	}); err != nil {
+		return err
+	}
+
+	return err
```
[the remark this answers is not in the document]
Comment (frrist) [reply pkg/downloader/s3/download.go:78]: ahh yes, thanks.

**Comment by wdbaruni** [comment]
Why do we think this is needed compared to using pre-signed urls by the requester node? I really doubt that in production we will have cases where the client has access to the bucket, but not the requester node

**Comment by frrist** [comment]
The team has stated this feature is too complicated to support. Closing.

**Pull request closed without merging**